pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -343,98 +539,72 @@ func CompactNewChunks(ctx context.Context, series Series, bt *v1.BloomTokenizer,
+
+		for _, meta := range metas {
+			for _, blockRef := range meta.Blocks {
+				uniqueIndexPaths[blockRef.IndexPath] = struct{}{}
+				// ...
+			}
+		}
 
-	if err != nil {
-		return err
 	}
 
-	for _, s := range series {
-		err := storeClient.indexShipper.ForEach(ctx, s.tableName, s.tenant, func(isMultiTenantIndex bool, idx shipperindex.Index) error {
-			if isMultiTenantIndex {
-				return nil
-			}
+	return nil
+}
 
-			// TODO make this casting safe
-			_ = idx.(*tsdb.TSDBFile).Index.(*tsdb.TSDBIndex).ForSeries(
-				ctx,
-				nil,              // Process all shards
-				0, math.MaxInt64, // Replace with MaxLookBackPeriod
-
-				// Get chunks for a series label and a fp
-				func(ls labels.Labels, fp model.Fingerprint, chksMetas []tsdbindex.ChunkMeta) {
-
-					// TODO call bloomShipperClient.GetMetas to get existing meta.json
-					var metas []bloomshipper.Meta
-
-					if len(metas) == 0 {
-						// Get chunks data from list of chunkRefs
-						chks, err := storeClient.chunk.GetChunks(
-							ctx,
-							makeChunkRefs(chksMetas, s.tenant, fp),
-						)
-						if err != nil {
-							level.Info(util_log.Logger).Log("getting chunks", err)
-							return
-						}
-
-						// effectively get min and max of timestamps of the list of chunks in a series
-						// There must be a better way to get this, ordering chunkRefs by timestamp doesn't fully solve it
-						// chunk files name have this info in ObjectStore, but it's not really exposed
-						minFrom := model.Latest
-						maxThrough := model.Earliest
-
-						for _, c := range chks {
-							if minFrom > c.From {
-								minFrom = c.From
-							}
-							if maxThrough < c.From {
-								maxThrough = c.Through
-							}
-						}
-
-						series := Series{
-							tableName:   s.tableName,
-							tenant:      s.tenant,
-							labels:      ls,
-							fingerPrint: fp,
-							chunks:      chks,
-							from:        minFrom,
-							through:     maxThrough,
-							indexPath:   s.indexPath,
-						}
-
-						err = CompactNewChunks(ctx, series, bt, bloomShipperClient, c.cfg.WorkingDirectory)
-						if err != nil {
-							return
-						}
-					} else {
-						// TODO complete part 2 - periodic compaction for delta from previous period
-						// When already compacted metas exists
-						// Deduplicate index paths
-						uniqueIndexPaths := make(map[string]struct{})
-
-						for _, meta := range metas {
-							for _, blockRef := range meta.Blocks {
-								uniqueIndexPaths[blockRef.IndexPath] = struct{}{}
-								//...
-							}
-						}
-
-					}
-				})
-			return nil
-		})
-		if err != nil {
-			return errors.Wrap(err, "getting each series")
+// filterAndSortTablesByRange returns most-recent first sorted list of tables that are within the max age.
+func filterAndSortTablesByRange(tables []string, maxAge time.Duration) []string {
+	tableRanges := make(map[string]model.Interval, len(tables))
+	tablesToSort := make([]string, 0, len(tables))
+	maxAgeTime := model.Now().Add(-maxAge)
```

Comment:
How can we ensure we exclude un-compacted index tables from potentially last 15 mins? we agreed that it will be tricky to compact most recent uncompacted tables.

Reply:
Good call. I think we'll need to add a new limit `MinTableAge`? and look `End` time of the table's interval.

Reply:
Is there any other way of determining an un-compacted indexes I wonder? 
I've seen some tar files in the index bucket in gcp, something like `timestamp-ingester-12.tsdb.gz`, i assume these are un-compacted indexes. I'll check the code tomorrow to confirm. 
I know the underlying object client doesn't filter them out, returns them all. But maybe something else in the middle does.

Reply:
> Is there any other way of determining an un-compacted indexes I wonder?

I'm not sure. Let me know the output of your investigation. We can modify this in a followup PR.

Added MinTableAge defaulting to 1h for now.